xarray/core/computation.py

```diff
@@ -408,7 +416,26 @@ def apply_dict_of_variables_vfunc(
 
     result_vars = {}
     for name, variable_args in zip(names, grouped_by_name):
-        result_vars[name] = func(*variable_args)
+        # TODO: is this a reasonable to check for missing core dims? We check for a dims
+        # property to protect against the case where a numpy array is passed in.
+        if hasattr(variable_args[0], "dims") and set(
+            signature.all_input_core_dims
+        ) - set(variable_args[0].dims):
```

Comment:
Do we need to check all variable_args here?

Reply:
Yes! Now done